mt7620 crash since kernel > 6.6.45

### Describe the bug

crash log:
```
<1>[  276.393904] CPU 0 Unable to handle kernel paging request at virtual address 7036837c, epc == 7036837c, ra == 80004574
<4>[  276.415127] Oops[#1]:
<4>[  276.419646] CPU: 0 PID: 11573 Comm: jshn Tainted: G           O       6.6.50 #0
<4>[  276.434213] $ 0   : 00000000 00000001 00000002 806d1698
<4>[  276.444644] $ 4   : 7fce7d9f 00000002 ffffff83 82979e00
<4>[  276.455068] $ 8   : 00000000 00000000 00000001 81995965
<4>[  276.465489] $12   : 00000000 8076eccc 00000003 85d5a630
<4>[  276.475911] $16   : 7fce7d9f 81995960 00000000 85d83e00
<4>[  276.486334] $20   : 85c8ac80 806d07a0 82ae3640 807b0000
<4>[  276.496758] $24   : 00000000 843b1cc4                  
<4>[  276.507180] $28   : 843b0000 843b1dd8 843b1df0 80004574
<4>[  276.517606] Hi    : 00000002
<4>[  276.523329] Lo    : 00000003
<4>[  276.529053] epc   : 7036837c 0x7036837c
<4>[  276.536692] ra    : 80004574 arch_align_stack+0x50/0x70
<4>[  276.547115] Status: 1100e403	KERNEL EXL IE 
<4>[  276.555454] Cause : 10800008 (ExcCode 02)
<4>[  276.563432] BadVA : 7036837c
<4>[  276.569156] PrId  : 00019650 (MIPS 24KEc)
<4>[  276.577133] Modules linked in: ksmbd rt2800soc(O) rt2800mmio(O) rt2800lib(O) qcserial pppoe ppp_async option cdc_mbim wireguard usb_wwan sierra_net sierra rt2x00soc(O) rt2x00mmio(O) rt2x00lib(O) rndis_host qmi_wwan pptp pppox ppp_mppe ppp_generic nft_fib_inet mac80211(O) libchacha20poly1305 ipt_REJECT huawei_cdc_ncm cfg80211(O) cdc_ncm cdc_ether xt_time xt_tcpudp xt_tcpmss xt_statistic xt_state xt_recent xt_nat xt_multiport xt_mark xt_mac xt_limit xt_length xt_hl xt_helper xt_ecn xt_dscp xt_conntrack xt_connmark xt_connlimit xt_connbytes xt_comment xt_TCPMSS xt_REDIRECT xt_MASQUERADE xt_LOG xt_HL xt_DSCP xt_CLASSIFY usbserial usbnet usblp ums_usbat ums_sddr55 ums_sddr09 ums_karma ums_jumpshot ums_isd200 ums_freecom ums_datafab ums_cypress ums_alauda ts_fsm ts_bm tcp_scalable tcp_bbr slhc sch_cake r8152 poly1305_mips nft_tproxy nft_socket nft_reject_ipv6 nft_reject_ipv4 nft_reject_inet nft_reject_bridge nft_reject nft_redir nft_quota nft_queue nft_numgen nft_nat nft_meta_bridge nft_masq nft_log nft_limit nft_hash
<4>[  276.577771]  nft_fwd_netdev nft_fib_ipv6 nft_fib_ipv4 nft_fib nft_dup_netdev nft_ct nft_compat nft_chain_nat nfnetlink_queue nf_tproxy_ipv6 nf_tproxy_ipv4 nf_tables nf_socket_ipv6 nf_socket_ipv4 nf_reject_ipv4 nf_nat_tftp nf_nat_snmp_basic nf_nat_sip nf_nat_pptp nf_nat_irc nf_nat_h323 nf_nat_ftp nf_nat_amanda nf_log_syslog nf_dup_netdev nf_conntrack_tftp nf_conntrack_snmp nf_conntrack_sip nf_conntrack_sane nf_conntrack_pptp nf_conntrack_netbios_ns nf_conntrack_irc nf_conntrack_h323 nf_conntrack_ftp nf_conntrack_broadcast nf_conntrack_bridge ts_kmp nf_conntrack_amanda nf_conncount macvlan libcurve25519_generic libcrc32c ipvlan iptable_nat iptable_mangle iptable_filter ipt_ECN ipheth ip_tables crc_ccitt compat(O) chacha_mips cdc_wdm br_netfilter natflow(O) natcap(O) nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 fuse ntfs3 sch_tbf sch_ingress sch_htb sch_hfsc em_u32 cls_u32 cls_route cls_matchall cls_fw cls_flow cls_basic act_skbedit act_mirred act_gact ledtrig_usbport xt_set ip_set_list_set ip_set_hash_netportnet
<4>[  276.756961]  ip_set_hash_netport ip_set_hash_netnet ip_set_hash_netiface ip_set_hash_net ip_set_hash_mac ip_set_hash_ipportnet ip_set_hash_ipportip ip_set_hash_ipport ip_set_hash_ipmark ip_set_hash_ipmac ip_set_hash_ip ip_set_bitmap_port ip_set_bitmap_ipmac ip_set_bitmap_ip ip_set nfnetlink ip6table_mangle ip6table_filter ip6_tables ip6t_REJECT x_tables nf_reject_ipv6 nfsv4 nls_ucs2_utils cifs_arc4 asn1_decoder nfs msdos ip6_gre ip_gre gre ip6_udp_tunnel udp_tunnel sit ip6_tunnel oid_registry tunnel6 tunnel4 ip_tunnel tun lockd sunrpc grace autofs4 dns_resolver nls_utf8 nls_iso8859_1 nls_cp437 crypto_user algif_skcipher algif_rng algif_hash algif_aead af_alg sha512_generic sha256_generic libsha256 sha1_generic seqiv sha3_generic drbg md5 hmac geniv rng ecb des_generic libdes cmac arc4 uas mtk_sd mmc_block cqhci usb_storage mmc_core leds_gpio xhci_plat_hcd xhci_pci xhci_hcd ohci_platform ohci_hcd fsl_mph_dr_of ehci_platform ehci_fsl sd_mod scsi_mod scsi_common ehci_hcd gpio_button_hotplug(O) vfat fat ext4 mbcache jbd2
<4>[  276.936754]  exfat usbcore nls_base usb_common mii crc32c_generic [last unloaded: ksmbd]
<4>[  277.132417] Process jshn (pid: 11573, threadinfo=71d86be0, task=a21c8eab, tls=77ef9df4)
<4>[  277.148373] Stack : 77e850ff 77dc6bc0 85d83e00 85c0c400 85d83800 801d39d4 00000006 00000000
<4>[  277.165055]         85d8385c 80820000 00000012 00000000 00000003 0041ff14 00401d50 00400000
<4>[  277.181732]         00420028 00400034 00000008 00401140 00000022 00000002 77dc6bc0 806d1698
<4>[  277.198409]         77db4000 85d83800 00000034 00000000 00000000 00000000 82ae36e0 00000000
<4>[  277.215084]         00000100 ecbcfece 808241dc fffffff8 85d83800 80823890 00000006 00000000
<4>[  277.231762]         ...
<4>[  277.236629] Call Trace:
<4>[  277.236634] 
<4>[  277.244604] [<801d39d4>] load_elf_binary+0xc44/0x1590
<4>[  277.254839] [<80174d54>] bprm_execve+0x1ec/0x578
<4>[  277.264053] [<801747a4>] copy_string_kernel+0x104/0x240
<4>[  277.274472] [<801756f0>] do_execveat_common+0x1b4/0x240
<4>[  277.284884] [<8000fdec>] do_page_fault+0xd4/0x554
<4>[  277.294419] [<80176378>] sys_execve+0x34/0x48
<4>[  277.303111] [<8000dee0>] syscall_common+0x34/0x58
<4>[  277.312502] 
<4>[  277.315456] Code: (Bad address in epc)
<4>[  277.315456] 
<4>[  277.325856] 
<4>[  277.329048] ---[ end trace 0000000000000000 ]---
```

no crash for `kernel version <= 6.6.45` 

### OpenWrt version

r27609-cf79b823996

### OpenWrt release

24.04_b202409140123

### OpenWrt target/subtarget

ramips/mt7620

### Device

Youku YK-L1c

### Image kind

Self-built image

### Steps to reproduce

_No response_

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

_No response_

### Diffconfig

_No response_

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

```
drivers/char/random.c:492:45: error: 'batched_entropy_u64' defined but not used [-Werror=unused-variable]
  492 | static DEFINE_PER_CPU(struct batch_ ##type, batched_entropy_ ##type) = {        \
      |                                             ^~~~~~~~~~~~~~~~
./include/linux/percpu-defs.h:104:44: note: in definition of macro 'DEFINE_PER_CPU_SECTION'
  104 |         __PCPU_ATTRS(sec) __typeof__(type) name
      |                                            ^~~~
drivers/char/random.c:492:8: note: in expansion of macro 'DEFINE_PER_CPU'
  492 | static DEFINE_PER_CPU(struct batch_ ##type, batched_entropy_ ##type) = {        \
      |        ^~~~~~~~~~~~~~
drivers/char/random.c:506:1: note: in expansion of macro 'DEFINE_BATCHED_ENTROPY'
  506 | DEFINE_BATCHED_ENTROPY(u64)
      | ^~~~~~~~~~~~~~~~~~~~~~
```
Repeating for `batched_entropy_u32`, `batched_entropy_u16`, `batched_entropy_u8`. Not sure how to pass ignore arguments down to kernels gcc. Is there any global OpenWRT build toggle I can change? Thanks

--- a/target/linux/ramips/patches-6.12/330-mt7620-avoid-direct-jal-in-random-helper.patch
+++ b/target/linux/ramips/patches-6.12/330-mt7620-avoid-direct-jal-in-random-helper.patch
@@ -1,0 +1,72 @@
+From: Mieczyslaw Nalewaj <namiltd@yahoo.com>
+Date: Sun, 26 Oct 2025 10:36:02 +0100
+Subject: [PATCH] mt7620: avoid direct jal imm26 in random helper
+
+This patch forces an indirect call (via a function pointer) on the mt7620
+platform when calling the small random helpers used by the constant-folding
+fast path of get_random_u32_below(). That avoids embedding a 26-bit immediate
+jump target in a direct `jal` instruction, which on affected platforms can
+lead to sporadic incorrect jump targets, intermittent oopses/crashes/hangs,
+corrupted backtraces, or "half-written" instructions under some circumstances.
+
+The change is targetted and conservative: it only alters code generation on
+CONFIG_NET_RALINK_MT7620 builds and does not change the algorithmic behaviour
+of get_random_u32_below().
+
+Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
+---
+ a/include/linux/random.h | 36 ++++++++++++++++++++++++++++++++++++----
+ 1 file changed, 32 insertions(+), 4 deletions(-)
+
+--- a/include/linux/random.h
++++ b/include/linux/random.h
+@@ -73,17 +73,45 @@ static inline u32 get_random_u32_below(u
+ 	BUILD_BUG_ON_MSG(!ceil, "get_random_u32_below() must take ceil > 0");
+ 	if (ceil <= 1)
+ 		return 0;
+-	for (;;) {
+-		if (ceil <= 1U << 8) {
++	if (ceil <= 1U << 8) {
++#ifdef CONFIG_NET_RALINK_MT7620
++		typedef u8 (*getrnd_t)(void);
++		volatile getrnd_t rnd_fn = (getrnd_t)get_random_u8;
++#endif
++		for (;;) {
++#ifdef CONFIG_NET_RALINK_MT7620
++			u32 mult = ceil * rnd_fn();
++#else
+ 			u32 mult = ceil * get_random_u8();
++#endif
+ 			if (likely(is_power_of_2(ceil) || (u8)mult >= (1U << 8) % ceil))
+ 				return mult >> 8;
+-		} else if (ceil <= 1U << 16) {
++		}
++	} else if (ceil <= 1U << 16) {
++#ifdef CONFIG_NET_RALINK_MT7620
++		typedef u16 (*getrnd_t)(void);
++		volatile getrnd_t rnd_fn = (getrnd_t)get_random_u16;
++#endif
++		for (;;) {
++#ifdef CONFIG_NET_RALINK_MT7620
++			u32 mult = ceil * rnd_fn();
++#else
+ 			u32 mult = ceil * get_random_u16();
++#endif
+ 			if (likely(is_power_of_2(ceil) || (u16)mult >= (1U << 16) % ceil))
+ 				return mult >> 16;
+-		} else {
++		}
++	} else {
++#ifdef CONFIG_NET_RALINK_MT7620
++		typedef u32 (*getrnd_t)(void);
++		volatile getrnd_t rnd_fn = (getrnd_t)get_random_u32;
++#endif
++		for (;;) {
++#ifdef CONFIG_NET_RALINK_MT7620
++			u64 mult = (u64)ceil * rnd_fn();
++#else
+ 			u64 mult = (u64)ceil * get_random_u32();
++#endif
+ 			if (likely(is_power_of_2(ceil) || (u32)mult >= -ceil % ceil))
+ 				return mult >> 32;
+ 		}

--- a/target/linux/ramips/patches-6.12/331-mt7620-avoid-direct-jal-in-resume_user_mode_work.patch
+++ b/target/linux/ramips/patches-6.12/331-mt7620-avoid-direct-jal-in-resume_user_mode_work.patch
@@ -1,0 +1,35 @@
+From: Mieczyslaw Nalewaj <namiltd@yahoo.com>
+Date: Tue, 28 Oct 2025 22:50:41 +0100
+Subject: [PATCH] mt7620: avoid direct jal imm26 in resume_user_mode_work
+
+This patch forces an indirect call (via a function pointer) on the mt7620
+platform when invoking blkcg_maybe_throttle_current() from
+resume_user_mode_work(). Avoiding a direct `jal` with a 26-bit immediate
+jump target prevents sporadic incorrect jump targets / intermittent oopses,
+crashes or hanging behavior that can appear on affected platforms.
+
+The change is targeted: it only alters the call site when CONFIG_NET_RALINK_MT7620
+is enabled and does not change the functional behaviour of resume_user_mode_work().
+
+Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
+
+---
+ a/include/linux/resume_user_mode.h | 6 +++++-
+ 1 file changed, 5 insertions(+), 1 deletion(-)
+
+--- a/include/linux/resume_user_mode.h
++++ b/include/linux/resume_user_mode.h
+@@ -57,7 +57,13 @@ static inline void resume_user_mode_work
+ #endif
+ 
+ 	mem_cgroup_handle_over_high(GFP_KERNEL);
++#ifdef CONFIG_NET_RALINK_MT7620
++	typedef void (*bmtc_t)(void);
++	volatile bmtc_t bmtc_fn = (bmtc_t)blkcg_maybe_throttle_current;
++	bmtc_fn();
++#else
+ 	blkcg_maybe_throttle_current();
++#endif
+ 
+ 	rseq_handle_notify_resume(NULL, regs);
+ }

--- a/target/linux/ramips/patches-6.12/332-mt7620-conditional-stack-align.patch
+++ b/target/linux/ramips/patches-6.12/332-mt7620-conditional-stack-align.patch
@@ -1,0 +1,30 @@
+From: Mieczyslaw Nalewaj <namiltd@yahoo.com>
+Date: Sun, 26 Oct 2025 10:36:02 +0100
+Subject: [PATCH] mt7620: conditional stack align
+
+This patch avoids applying arch_align_stack() and PAGE_ALIGN() in the exec
+path on CONFIG_NET_RALINK_MT7620 builds. Many ports (including mt7620)
+implement only micro-randomization inside arch_align_stack() (random offset
+< PAGE_SIZE and then align-down to small alignment). Callers that immediately
+apply PAGE_ALIGN() will round that micro-offset back to the original page
+boundary, so invoking arch_align_stack() then PAGE_ALIGN() is pointless and
+can be avoided on the affected platform.
+
+Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
+---
+ a/fs/exec.c | 2 ++
+ 1 file changed, 2 insertions(+)
+
+--- a/fs/exec.c
++++ b/fs/exec.c
+@@ -750,8 +750,10 @@ int setup_arg_pages(struct linux_binprm
+ 	mm->arg_start = bprm->p - stack_shift;
+ 	bprm->p = vma->vm_end - stack_shift;
+ #else
++#ifndef CONFIG_NET_RALINK_MT7620
+ 	stack_top = arch_align_stack(stack_top);
+ 	stack_top = PAGE_ALIGN(stack_top);
++#endif
+ 
+ 	if (unlikely(stack_top < mmap_min_addr) ||
+ 	    unlikely(vma->vm_end - vma->vm_start >= stack_top - mmap_min_addr))

